### PR TITLE
refactor(ledger): shred field renames

### PR DESCRIPTION
**Pull request by dnut**
Previously there was a mix of items named "code shred" and "coding shred". This change settles on "code shred" as the name to use everywhere. They are interchangeable so consistency is the priority here.

I also renamed variables and fields called "shred_variant" to "variant" when they are in the scope of a shred

### CODE REVIEW

File: src/ledger/meta.zig [hunk 1]
```diff
@@ -4,7 +4,7 @@ const sig = @import("../lib.zig");
 const Allocator = std.mem.Allocator;
 
 const BitFlags = sig.utils.bitflags.BitFlags;
-const CodingShred = sig.ledger.shred.CodingShred;
+const CodingShred = sig.ledger.shred.CodeShred;
```
Comment (dadepo) [inline src/ledger/meta.zig:7]: Should the type also be renamed to `CodeShred` here?

File: src/ledger/meta.zig [hunk 2]
```diff
@@ -190,18 +190,18 @@ pub const ErasureMeta = struct {
...
-            .first_received_coding_index = @intCast(shred.fields.common.index),
+            .first_code_index = @intCast(shred.firstCodingIndex() catch return null),
+            .first_received_code_index = @intCast(shred.fields.common.index),
         };
     }
 
     /// Returns true if the erasure fields on the shred
     /// are consistent with the erasure-meta.
     pub fn checkCodingShred(self: Self, shred: CodingShred) bool {
         var other = fromCodingShred(shred) orelse return false;
```
Comment (dadepo) [inline src/ledger/meta.zig:203]: should the `fn` also be renamed to `fromCodeShred`?

File: src/ledger/shredder.zig [hunk 3]
```diff
@@ -3,8 +3,8 @@ const sig = @import("../lib.zig");
 
 const Allocator = std.mem.Allocator;
 
-const CodingShred = sig.ledger.shred.CodingShred;
-const CodingShredHeader = sig.ledger.shred.CodingShredHeader;
+const CodingShred = sig.ledger.shred.CodeShred;
```
Comment (dadepo) [inline src/ledger/shredder.zig:6]: should this be also be renamed to `const CodeShred` and `CodeShredHeader`?

**Comment by dadepo** [comment]
Looks the renaming was missed here https://github.com/Syndica/sig/blob/5365f1fe450cd8b004360a76a4305e83293b1cfe/src/ledger/cleanup_service.zig#L201

File: src/ledger/insert_shred.zig [hunk 4]
```diff
@@ -207,10 +207,10 @@ pub const ShredInserter = struct {
                         else => return e, // TODO explicit
                     }
                 },
-                .code => |coding_shred| {
+                .code => |code_shred| {
                     // TODO error handling?
                     _ = try self.checkInsertCodingShred(
```
Comment (dadepo) [inline src/ledger/insert_shred.zig:212]: Looks like the method name can also be renamed? `checkInsertCodeShred`?

File: src/ledger/insert_shred.zig [hunk 5]
```diff
@@ -493,7 +493,7 @@ pub const ShredInserter = struct {
         // agave runs this regardless of trust, but we can check if it has
         // a meaningful performance impact to skip this for trusted shreds.
         if (!erasure_meta.checkCodingShred(shred)) {
```
Comment (dadepo) [inline src/ledger/insert_shred.zig:495]: The method can also be renamed?

File: src/ledger/insert_shred.zig [hunk 6]
```diff
@@ -493,7 +493,7 @@ pub const ShredInserter = struct {
...
-            self.metrics.num_coding_shreds_invalid_erasure_config.inc();
+            self.metrics.num_code_shreds_invalid_erasure_config.inc();
             if (!try self.hasDuplicateShredsInSlot(slot)) {
                 if (try self.findConflictingCodingShred(
```
Comment (dadepo) [inline src/ledger/insert_shred.zig:498]: Here also. the `findConflictingCodingShred` method

**Comment by dnut** [comment]
@dadepo I have addressed all your comments

**Pull request closed without merging**